pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -129,10 +129,18 @@ func (dswp *desiredStateOfWorldPopulator) populatorLoopFunc() func() {
 	}
 }
 
+func isPodTerminated(pod *api.Pod) bool {
+	return pod.Status.Phase == api.PodFailed || pod.Status.Phase == api.PodSucceeded
+}
+
 // Iterate through all pods and add to desired state of world if they don't
 // exist but should
 func (dswp *desiredStateOfWorldPopulator) findAndAddNewPods() {
 	for _, pod := range dswp.podManager.GetPods() {
+		if isPodTerminated(pod) {
+			// Do not (re)add volumes for terminated pods
```

Comment:
@saad-ali the change required to check that actually makes the change more invasive because I need to get the volume.  In `findAndRemoveDeletedPods()`, that is already available via `volumeToMount`.  In `findAndAddNewPods()`, the logic is inverted; for each pod, process volumes.  I'm not sure if it is safe to call `dswp.desiredStateOfWorld.GetVolumesToMount()` in `findAndAddNewPods()`.

I can extend my e2e tests to include a check that ensures non-memory backed volumes are untouched by this change?